include/numerics/type_vector.h

```diff
@@ -1100,6 +1100,31 @@ T cross_norm(const TypeVector<T> & b,
   return std::sqrt(cross_norm_sq(b,c));
 }
 
+template <typename T>
+inline
+TypeVector<T> TypeVector<T>::unit() const
+{
+
+  auto && length = norm();
```

Comment:
You haven't changed `norm()`? So that's always returning `Real` regardless of `T`... maybe you plan to change that later, but I am curious about the use of `auto &&` instead of simply `auto`? Does it avoid a copy or something?

Reply:
`norm()` is changed in #1946 (which should be much better off after idaholab/moose#12599). Since `norm` just returns a value, the addition of `&&` makes no difference. I guess I've just been getting in the habit of writing this kind of general code to, in general, avoid copies.